main.cpp

```diff
@@ -421,6 +417,27 @@ DoubleType& DoubleType::pow(const IntType& it)
     return powInternal(static_cast<double>(it));
 }
 
+DoubleType& DoubleType::apply(std::function<DoubleType&(double&)> func)
+{
+    if(func)
+        func(*value);
```

Comment:
the lambda returns a reference to `DoubleType&`. 
This function also returns a reference to `DoubleType&`. 
```
 2) One of the apply() functions should takes a std::function<> object as the function argument.
    the std::function<> object should return *this;
```
if `func` is valid, then return the result that `func` returns from this function. 
- [ ] revise